### PR TITLE
Begin broker/client config messaging.

### PR DESCRIPTION
**Pull request by corylanou**
WIP, nothing to see yet

### CODE REVIEW

**Comment by corylanou** [comment]
@benbjohnson can you give this a glance and see if it's ok to merge.  @levicook could use this.

File: cmd/influxd/run.go [hunk 1]
```diff
@@ -249,6 +257,24 @@ func initializeServer(s *influxdb.Server, b *messaging.Broker) {
 	}
 }
 
+func openDefaultServer(s *influxdb.Server, b *messaging.Broker) {
+	// TODO: Create replica using the messaging client.
+
+	// Create replica on broker.
+	if err := b.CreateReplica(1); err != nil {
```
Comment (pauldix) [inline cmd/influxd/run.go]: what is the `1` being passed into `CreateReplica`? This is an entirely new cluster so `1` is the id of the server?

Comment (corylanou) [inline cmd/influxd/run.go]: I'm not 100% sure.  I had a lot of questions on what some of this code did so I made some assumptions...  Hopefully Ben can shed some light on this.

Comment (pauldix) [inline cmd/influxd/run.go]: I'm pretty sure that's the thing.

**Comment by pauldix** [comment]
LGTM, I say :ship:, but maybe wait for @benbjohnson's feedback?

File: cmd/influxd/run.go [hunk 2]
```diff
@@ -218,6 +222,10 @@ func openServer(path string, u *url.URL, b *messaging.Broker, initializing bool,
 			joinServer(s, u, joinURLs)
 			openServerClient(s, joinURLs)
 		}
+	} else if !configExists {
+		// We are spining up an server that has no config,
+		// but already has an initialized data directory
+		openDefaultServer(s, b)
```
Comment (benbjohnson) [inline cmd/influxd/run.go]: You shouldn't need to duplicate the replica and server creation. If there's no config then should just need to set `joinURLs` to `[]*url.URL{b.URL}` and call `openServerClient(s, joinURLs)`.

File: cmd/influxd/run.go [hunk 3]
```diff
@@ -133,11 +134,14 @@ func writePIDFile(path string) {
 
 // parses the configuration from a given path. Sets overrides as needed.
 func parseConfig(path, hostname string) *Config {
+	if path == "" {
+		log.Println("No config provided, using default settings")
```
Comment (otoolep) [inline cmd/influxd/run.go]: +1 for log message.

File: cmd/influxd/run.go [hunk 4]
```diff
@@ -198,7 +202,7 @@ func joinBroker(b *messaging.Broker, joinURLs []*url.URL) {
 }
 
 // creates and initializes a server.
-func openServer(path string, u *url.URL, b *messaging.Broker, initializing bool, joinURLs []*url.URL) *influxdb.Server {
+func openServer(path string, u *url.URL, b *messaging.Broker, initializing, configExists bool, joinURLs []*url.URL) *influxdb.Server {
```
Comment (otoolep) [inline cmd/influxd/run.go]: Minor: I do think this `configExists` logic should have been executed *before* the call to `openServer()`. That way `openServer()` remained untouched. Right now it seems a little off that `configExists` has to be pushed into the function.

I.e.

```
// Special case if no config file was specified.
if !configExists {
    joinURLs = []*url.URL{b.URL()}
}
.
.
.
s := openServer(config.Data.Dir, config.DataURL(), b, initializing, joinURLs)
```
This seems like clearer code to me.

**Pull request merged**